internal/app/machined/pkg/system/runner/process/process.go

```diff
@@ -206,20 +322,29 @@ func (p *processRunner) run(eventSink events.Recorder) error {
 		defer reaper.Stop(notifyCh)
 	}
 
-	err = cmdWrapper.cmd.Start()
+	pid, err := cmdWrapper.launcher.Launch(&cmdWrapper)
+	if err != nil {
+		return fmt.Errorf("error starting process: %w", err)
+	}
+
+	if err := applyProperties(p, pid); err != nil {
+		return err
+	}
 
 	cmdWrapper.afterStart()
 
+	eventSink(events.StateRunning, "Process %s started with PID %d", p, pid)
+
+	process, err := os.FindProcess(pid)
 	if err != nil {
-		return fmt.Errorf("error starting process: %w", err)
+		return fmt.Errorf("could not find process: %w", err)
 	}
 
-	eventSink(events.StateRunning, "Process %s started with PID %d", p, cmdWrapper.cmd.Process.Pid)
-
 	waitCh := make(chan error)
 
 	go func() {
-		waitCh <- reaper.WaitWrapper(usingReaper, notifyCh, cmdWrapper.cmd)
+		_, err := process.Wait()
```

Comment:
should use reaper, don't change this part please

Reply:
Okay, refactoring here to accept raw pid and so: https://github.com/siderolabs/go-cmd/blob/main/pkg/cmd/proc/reaper/wait.go#L13-L38